mail_notification_with_history/models/mail_message.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import models
+
+
+class Message(models.Model):
+    _inherit = "mail.message"
+
+    def get_notification_message_history(self):
+        """Get the list of messages to include into an email notification history."""
+        if not self.env[self.model].with_mail_notification_history():
+            return []
+        if not self._with_notification_message_history():
+            return []
```

Comment:
```suggestion
        if (
            not self.env[self.model].with_mail_notification_history()
            or not self._with_notification_message_history()
        ):
            return self.browse()
```

note that, since you'd normally return a `mail.message` recordset, here we should return an empty recordset for consistency rather than an empty list.

Reply:
Although, as per my comment below, I'd go for:

```suggestion
        if (
            not self.env[self.model]._mail_notification_include_history
            or not self._with_notification_message_history()
        ):
            return self.browse()
```

Reply:
Yes for the browse instead of the list.